validator/src/main/java/com/google/bos/iot/core/bambi/BambiSiteModelManager.java

```diff
@@ -0,0 +1,295 @@
+    try {
+      spreadsheetManager = new SpreadsheetManager("BAMBI", spreadsheetId);
+      bambiSiteModel = new BambiSiteModel(
+          getRecordsFromSheet(BambiSheet.SITE_METADATA),
+          getRecordsFromSheet(BambiSheet.CLOUD_IOT_CONFIG),
+          getRecordsFromSheet(BambiSheet.SYSTEM),
+          getRecordsFromSheet(BambiSheet.CLOUD),
+          getRecordsFromSheet(BambiSheet.GATEWAY),
+          getRecordsFromSheet(BambiSheet.LOCALNET),
+          getRecordsFromSheet(BambiSheet.POINTSET),
+          getRecordsFromSheet(BambiSheet.POINTS)
+      );
+    } catch (IOException e) {
+      throw new RuntimeException("while initializing BambiSiteModelManager: ", e);
+    }
+  }
+
+  public BambiSiteModel getBambiSiteModel() {
+    return bambiSiteModel;
+  }
+
+  public Map<String, String> getSiteMetadata() {
+    return bambiSiteModel.getSiteMetadata();
+  }
+
+  public Map<String, String> getCloudIotConfig() {
+    return bambiSiteModel.getCloudIotConfig();
+  }
+
+  public Map<String, Map<String, String>> getAllDevicesMetadata() {
+    return bambiSiteModel.getAllDevicesMetadata();
+  }
+
+  /**
+   * Write relevant data to the site_metadata tab in the spreadsheet.
+   *
+   * @param newSiteMetadata Map of data to write
+   */
+  public void writeSiteMetadata(Map<String, String> newSiteMetadata) {
+    writeKeyValueTypeMetadata(BambiSheet.SITE_METADATA, bambiSiteModel.getSiteMetadataHeaders(),
+        newSiteMetadata);
+  }
+
+  /**
+   * Write relevant data to the cloud_iot_config tab in the spreadsheet.
+   *
+   * @param newCloudIotConfig Map of data to write
+   */
+  public void writeCloudIotConfig(Map<String, String> newCloudIotConfig) {
+    writeKeyValueTypeMetadata(BambiSheet.CLOUD_IOT_CONFIG,
+        bambiSiteModel.getCloudIotConfigHeaders(), newCloudIotConfig);
+  }
+
+  private void writeKeyValueTypeMetadata(BambiSheet sheet, List<String> headers,
+      Map<String, String> data) {
+    LOGGER.info("writing to sheet " + sheet.getName());
+    Map<String, String> newData = new LinkedHashMap<>();
+    for (String header : headers) {
+      newData.put(header, data.getOrDefault(header, ""));
+    }
+    try {
+      spreadsheetManager.clearValuesFromRange(sheet.getName());
+      spreadsheetManager.writeToRange(sheet.getName(), asSheetData(newData));
+    } catch (IOException e) {
+      throw new RuntimeException("while writing to bambi sheet " + sheet.getName(), e);
+    }
+  }
+
+  /**
+   * Write relevant data to system, cloud, gateway, localnet, pointset, and points tabs in the
+   * spreadsheet.
+   *
+   * @param deviceToMetadataMap Mapping of device id to the device metadata
+   */
+  public void writeDevicesMetadata(Map<String, Map<String, String>> deviceToMetadataMap) {
+    Map<BambiSheet, List<List<Object>>> dataToWrite = new HashMap<>();
+
+    List<SheetConfig> simpleTableConfigs = List.of(
+        new SheetConfig(BambiSheet.SYSTEM, bambiSiteModel.getSystemDataHeaders(), "system."),
```

Comment:
seems like the last argument ("system.") can/should be implicitly calculated from the first (BambiSheet.SYSTEM)?